File: NickLodewijks/src/sc/ql/ast/expr/StringLiteral.java
package sc.ql.ast.expr;

import sc.ql.ast.value.StringValue;

public final class StringLiteral extends Expr {

	private final StringValue value;

	public StringLiteral(String value) {
		this.value = new StringValue(value);
	}

	public StringValue getValue() {
		return value;
	}

	@Override
	public <T, U> T accept(ExprVisitor<T, U> visitor, U context) {
		return visitor.visit(this, context);
	}
}

File: Edward Poot/src/ql/ast/expr/stringliteral.go
package expr

import "ql/interfaces"

type StringLiteral struct {
	stringValue StringValue
	Expr
}

func NewStringLiteral(value string) StringLiteral {
	return StringLiteral{stringValue: NewStringValue(value), Expr: NewExpr()}
}

func (this StringLiteral) Value() interfaces.Value {
	return this.stringValue
}
